pages/instructorFileBrowser/instructorFileBrowser.js

```diff
@@ -202,89 +181,74 @@ function browseDirectory(file_browser, callback) {
+                isFile: true,
+                encodedName: encodePath(filename),
+                path: path.relative(file_browser.paths.coursePath, filepath),
+                encodedPath: encodePath(path.relative(file_browser.paths.coursePath, filepath)),
+                dir: file_browser.paths.workingPath,
+                canEdit:
+                  editable &&
+                  file_browser.has_course_permission_edit &&
+                  !file_browser.example_course,
+                canUpload: file_browser.has_course_permission_edit && !file_browser.example_course,
+                canDownload: true, // we already know the user is a course Viewer (checked on GET)
+                canRename:
+                  movable &&
+                  file_browser.has_course_permission_edit &&
+                  !file_browser.example_course,
+                canDelete:
+                  movable &&
+                  file_browser.has_course_permission_edit &&
+                  !file_browser.example_course,
+                canView: !file_browser.paths.invalidRootPaths.some((invalidRootPath) =>
+                  contains(invalidRootPath, filepath)
+                ),
+              };
+            } else if (stats.isDirectory()) {
+              return {
+                id: index,
+                name: filename,
+                isDirectory: true,
+                encodedName: encodePath(filename),
+                path: path.relative(file_browser.paths.coursePath, filepath),
+                encodedPath: encodePath(path.relative(file_browser.paths.coursePath, filepath)),
+                canView: !file_browser.paths.invalidRootPaths.some((invalidRootPath) =>
+                  contains(invalidRootPath, filepath)
+                ),
+              };
+            } else {
+              return null;
+            }
           }
         );
+        file_browser.files = all_files.filter((f) => f?.isFile).sort((a, b) => a.name.localeCompare(b.name));
+        file_browser.dirs = all_files.filter((f) => f?.isDirectory).sort((a, b) => a.name.localeCompare(b.name));
       },
-      (callback) => {
-        async.eachOfSeries(
-          file_browser.files,
-          (file, index, callback) => {
-            util.callbackify(editorUtil.getErrorsAndWarningsForFilePath)(
-              file_browser.paths.courseId,
-              file.path,
-              (err, data) => {
-                if (ERR(err, callback)) return;
-                const ansiUp = new AnsiUp();
-                file.sync_errors = data.errors;
-                file.sync_errors_ansified = ansiUp.ansi_to_html(file.sync_errors);
-                file.sync_warnings = data.warnings;
-                file.sync_warnings_ansified = ansiUp.ansi_to_html(file.sync_warnings);
-                callback(null);
-              }
-            );
-          },
-          (err) => {
-            if (ERR(err, callback)) return;
-            callback(null);
-          }
-        );
+      async () => {
```

Comment:
Minor nit: we could collapse these into a single `async () => { ... }` function; now that we're not using callbacks, there's no need for them to be separate.

Reply:
Done in 946fdd6.